test/framework/src/main/java/org/elasticsearch/repositories/AbstractThirdPartyRepositoryTestCase.java

```diff
@@ -218,6 +224,58 @@ public void testCleanup() throws Exception {
         assertCleanupResponse(response, 3L, 1L);
     }
 
+    public void testIndexLatest() throws Exception {
+        // This test verifies that every completed snapshot operation updates a blob called literally 'index.latest' (by default at least),
+        // which is important because some external systems use the freshness of this specific blob as an indicator of whether a repository
+        // is in use. Most notably, ESS checks this blob as an extra layer of protection against a bug in the delete-old-repositories
+        // process incorrectly deleting repositories that have seen recent writes. It's possible that some future development might change
+        // the meaning of this blob, and that's ok, but we must continue to update it to keep those external systems working.
+
+        createIndex("test-idx-1");
+        for (int i = 0; i < 100; i++) {
+            client().prepareIndex("test-idx-1").setId(Integer.toString(i)).setSource("foo", "bar" + i).get();
+        }
+
+        final var repository = getRepository();
+        final var blobContents = new HashSet<BytesReference>();
+
+        final var createSnapshot1Response = clusterAdmin().prepareCreateSnapshot(TEST_REPO_NAME, randomIdentifier())
+            .setWaitForCompletion(true)
+            .get();
+        assertTrue(blobContents.add(readIndexLatest(repository)));
+
+        clusterAdmin().prepareGetSnapshots(TEST_REPO_NAME).get();
+        assertFalse(blobContents.add(readIndexLatest(repository)));
+
+        final var createSnapshot2Response = clusterAdmin().prepareCreateSnapshot(TEST_REPO_NAME, randomIdentifier())
+            .setWaitForCompletion(true)
+            .get();
+        assertTrue(blobContents.add(readIndexLatest(repository)));
+
+        assertAcked(clusterAdmin().prepareDeleteSnapshot(TEST_REPO_NAME, createSnapshot1Response.getSnapshotInfo().snapshotId().getName()));
+        assertTrue(blobContents.add(readIndexLatest(repository)));
+
+        assertAcked(clusterAdmin().prepareDeleteSnapshot(TEST_REPO_NAME, createSnapshot2Response.getSnapshotInfo().snapshotId().getName()));
+        assertTrue(blobContents.add(readIndexLatest(repository)));
```

Comment:
Is it useful to have a cleanup call as well?

Reply:
I worry that we might make repo cleanup detect no-op runs better in future which would invalidate the assertion in this test that the `index.latest` blob changes. I think snapshot creation & deletion is enough.